log/namespaced/namespaces_test.go

```diff
@@ -0,0 +1,86 @@
+package namespaced
+
+import (
+	"testing"
+
+	. "github.com/smartystreets/assertions"
+)
+
+func TestNamespaces(t *testing.T) {
+	a := New(t)
+
+	ns := &ns{}
+
+	// empty namespace is accepted
+	{
+		ns.Set([]string{})
+
+		a.So(ns.IsEnabled("a"), ShouldBeFalse)
+		a.So(ns.IsEnabled("b"), ShouldBeFalse)
+		a.So(ns.IsEnabled("c"), ShouldBeFalse)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// add some namespaces
+	{
+		ns.Set([]string{
+			"a",
+			"b",
+		})
+
+		a.So(ns.IsEnabled("a"), ShouldBeTrue)
+		a.So(ns.IsEnabled("b"), ShouldBeTrue)
+		a.So(ns.IsEnabled("c"), ShouldBeFalse)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// * accepts everything
+	{
+		ns.Set([]string{
+			"*",
+		})
+
+		a.So(ns.IsEnabled("a"), ShouldBeTrue)
+		a.So(ns.IsEnabled("b"), ShouldBeTrue)
+		a.So(ns.IsEnabled("c"), ShouldBeTrue)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// negation wins from *
+	{
+		ns.Set([]string{
+			"*",
+			"-a",
+		})
+
+		a.So(ns.IsEnabled("a"), ShouldBeFalse)
+		a.So(ns.IsEnabled("b"), ShouldBeTrue)
+		a.So(ns.IsEnabled("c"), ShouldBeTrue)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// order should not matter
+	{
+		ns.Set([]string{
+			"-a",
+			"-b",
+			"*",
+		})
+
```

Comment:
Yes indeed I guess that if they are here it is to use them. However the natural way to exclude elements from a group is to specify the group and then to specify the elements to exclude. It's just a matter of reading "conventions" that I'm used to use. 

Here you adopted another convention. That's fine by me, just saying that I don't find it very natural and "conventional"